accounts/abi/bind/base.go

```
@@ -513,8 +515,10 @@ func (c *BoundContract) UnpackLog(out interface{}, event string, log types.Log)
 
 // UnpackLogIntoMap unpacks a retrieved log into the provided map.
 func (c *BoundContract) UnpackLogIntoMap(out map[string]interface{}, event string, log types.Log) error {
-	if log.Topics[0] != c.abi.Events[event].ID {
-		return fmt.Errorf("event signature mismatch")
+	if len(log.Topics) > 0 {
+		if log.Topics[0] != c.abi.Events[event].ID {
+			return fmt.Errorf("event signature mismatch")
```

same comment:
should we return an error in case `len(log.topics) < 1`?